src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart();
+
+        /// <summary>
+        /// Handle stop notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStop();
+    }
```

Comment:
For the purpose of information, I think it maybe worthwhile to add metadata field in the interface level to contain information of Stage and Observable it subscribes to. Otherwise it will move the burden to the application layer to keep those info around.